src/ddmd/func.d

```diff
@@ -3189,6 +3182,35 @@ extern (C++) final class UnitTestDeclaration : FuncDeclaration
         return FuncDeclaration.syntaxCopy(utd);
     }
 
+    /**
+       Sets the "real" identifier, replacing the one created in the contructor.
+       The reason for this is that the "real" identifier can only be generated
+       properly in the semantic pass. See:
+       https://issues.dlang.org/show_bug.cgi?id=16995
+     */
+    final void setIdentifier()
+    {
+        ident = createIdentifier(loc, _scope);
+    }
+
+    /***********************************************************
+     * Generate unique unittest function Id so we can have multiple
+     * instances per module.
+     */
+    private static Identifier createIdentifier(Loc loc, Scope* sc)
+    {
+        OutBuffer buf;
+        auto index = sc ? sc._module.unitTestCounter++ : 0;
+        buf.printf("__unittest_%s_%u_%d", loc.filename, loc.linnum, index);
```

Comment:
There's a difference without a distinction between %u and %d - just use %u for both, or better yet %s for all (assuming they're all unsigned as they should).

Reply:
`printf` doesn't work with `%s` for integers :-)

Reply:
Oh wait I thought it's writef...